utoipa-gen/src/path/media_type/encoding.rs

```diff
@@ -0,0 +1,92 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote, ToTokens};
+use syn::parse::{Parse, ParseStream};
+use syn::{parenthesized, Error, LitBool, LitStr, Token};
+
+use crate::parse_utils;
+
+// (content_type = "...", explode = true, allow_reserved = false,)
+#[derive(Default)]
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub struct Encoding {
+    pub(super) content_type: Option<String>,
+    // pub(super) headers: BTreeMap<String, Header>,
+    // pub(super) style: Option<ParameterStyle>,
+    pub(super) explode: Option<bool>,
+    pub(super) allow_reserved: Option<bool>,
+    // pub(super) extensions: Option<Extensions>,
+}
+
+impl Parse for Encoding {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let content;
+        parenthesized!(content in input);
+
+        let mut encoding = Encoding::default();
+
+        while !content.is_empty() {
+            let ident = content.parse::<Ident>()?;
+            let attribute_name = &*ident.to_string();
+            match attribute_name {
+                "content_type" => {
+                    encoding.content_type = Some(
+                        parse_utils::parse_next(&content, || content.parse::<LitStr>())?
+                            .value(),
+                    )
+                }
+                // "headers" => {}
+                // "style" => {}
+                "explode" => {
+                    encoding.explode = Some(
+                        parse_utils::parse_next(&content, || content.parse::<LitBool>())?
+                            .value(),
+                    )
+                }
+                "allow_reserved" => {
+                    encoding.allow_reserved = Some(
+                        parse_utils::parse_next(&content, || content.parse::<LitBool>())?.value()
+                    )
```

Comment:
These could use `parse_utils::parse_bool_or_true(...)`. This allows users to omit the `bool` value if true is expected. This behavior is already in many places of utoipa.

Reply:
Um wait, I am not going to do this yet because I am actually pretty certain the default for `explode` and `allow_reserved` would not be true, but I don't use them. If they are supposed to default to true I would be happy to change it.

Reply:
Nvm, did it.